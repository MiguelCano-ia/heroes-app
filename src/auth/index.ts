export * from './context';
export * from './interfaces';
export * from './pages';
export * from './types';