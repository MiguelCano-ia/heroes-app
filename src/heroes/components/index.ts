export * from './HeroList';
export * from './HeroList'