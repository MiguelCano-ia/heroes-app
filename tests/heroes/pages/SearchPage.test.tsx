import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import { SearchPage } from '../../../src/heroes';
import { MemoryRouter } from 'react-router-dom';

const mockedUseNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUseNavigate,
}));

describe('Pruebas en <SearchPage />', () => {

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('debe de mostrarse correctamente con valores por defecto', () => {
    
    const { container } =  render(
      <MemoryRouter>
        <SearchPage />
      </MemoryRouter>
    );

    expect( container ).toMatchSnapshot();
  });

  test('debe de mostrar a batman y el valor del queryString', () => {
    
    render(
      <MemoryRouter initialEntries={['/search?q=batman']} >
        <SearchPage />
      </MemoryRouter>
    );

    const input = screen.getByRole('textbox') as HTMLInputElement;
    expect( input.value ).toBe('batman');

    const img = screen.getByRole('img') as HTMLImageElement; ;
    expect( img.src ).toContain('/assets/heroes/dc-batman.jpg');

    const div = screen.getByLabelText('error-message');
    expect( div.style.display ).toBe('none');
  });

  test('debe de mostrar un error si no se encuentra el hero', () => {

    render(
      <MemoryRouter initialEntries={['/search?q=batman123']} >
        <SearchPage />
      </MemoryRouter>
    );

    const div = screen.getByLabelText('error-message');
    expect( div.style.display ).not.toBe('none');
  });

  test('debe de llamar el navegate a la pantalla nueva ', () => {
    
    const searchText = 'flash';
    const navigate = `?q=${ searchText }`;

    render(
      <MemoryRouter initialEntries={['/search']} >
        <SearchPage />
      </MemoryRouter>
    );

    const input = screen.getByRole('textbox') as HTMLInputElement;
    const form = screen.getByLabelText('form') as HTMLFormElement;

    fireEvent.input( input, { target: { value: searchText } });
    fireEvent.submit( form );

    expect( mockedUseNavigate ).toHaveBeenCalledWith( navigate );
  });
});
