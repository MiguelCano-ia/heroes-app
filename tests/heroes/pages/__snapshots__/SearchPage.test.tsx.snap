// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SearchPage /> debe de mostrarse correctamente con valores por defecto 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form>
        <input
          aria-label="form"
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Find your hero"
          type="text"
          value=""
        />
        <button
          class="btn mt-1 btn-outline-primary mb-5 mt-2"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        class="alert alert-info animate__animated animate__fadeIn"
      >
        Search a hero
      </div>
      <div
        aria-label="error-message"
        class="alert alert-danger animate__animated animate__fadeIn"
        style="display: none;"
      >
        There is no a hero with 
        <b />
      </div>
    </div>
  </div>
</div>
`;
